apps/src/netsim/NetSimRemoteNodeSelectionPanel.html.ejs

```diff
@@ -87,68 +80,131 @@ var requestRows = incomingConnectionNodes
-  <% } %>
+  }
+
+  markup += '>' + buttonText + '</span>';
+  return markup;
+}
+
+function writeHeader(headerText) {
+  %>
+    <tr>
+      <th colspan="3"><%= headerText %></th>
+    </tr>
+  <%
+}
+
+function writeEmptyRow(contents) {
+  contents = utils.valueOr(contents, '');
+  %>
+    <tr>
+      <td colspan="3" class="empty-row"><%- contents %></td>
+    </tr>
+  <%
+}
+
+function writeNodeRow(nodeID, nodeName, nodeStatus, buttonType, addlClass) {
+  %>
+    <tr>
+      <td nowrap><%= nodeName %></td>
+      <td><%- nodeStatus %></td>
+      <td class="button-column">
+        <%
+          var markup = '';
+          if (buttonType === 'join-button') {
+            markup = buttonMarkup(i18n.buttonJoin(), undefined, [buttonType, addlClass], { 'data-node-id': nodeID });
+          } else if (buttonType === 'accept-button') {
+            markup = buttonMarkup(i18n.buttonAccept(), undefined, [buttonType, addlClass], { 'data-node-id': nodeID });
+          } else if (buttonType === 'cancel-button') {
+            markup = buttonMarkup(i18n.buttonCancel(), undefined, [buttonType, addlClass, 'secondary'], { 'data-node-id': nodeID });
+          }
+        %>
+        <%- markup %>
+      </td>
+    </tr>
+  <%
+}
+
+%>
+<div class="content-wrap">
+  <div class="instructions"><%= controller.getLocalizedLobbyInstructions() %></div>
+  <div class="controls">
+    <table>
+
+      <%
+        // Primary lobby list
+        writeHeader(i18n.lobby());
+        lobbyRows.forEach(function (row) {
+          var buttonType;
+          if (!controller.hasOutgoingRequest() && row.canConnectToNode) {
+            buttonType = 'join-button';
+          }
+          writeNodeRow(row.nodeID, row.displayName, row.status, buttonType, row.classAttr);
+        });
+
+        if (!controller.hasOutgoingRequest() && showAddRouterButton) {
+          writeEmptyRow(buttonMarkup(i18n.addRouter(), 'netsim-lobby-add-router', ['secondary']));
+        } else if (lobbyRows.length === 0) {
+          writeEmptyRow(i18n.lobbyIsEmpty());
+        }
+
+        // Incoming requests table
+        if (requestRows.length > 0) {
+          writeEmptyRow();
+          writeHeader(i18n.incomingConnectionRequests());
+          requestRows.forEach(function (row) {
+            var buttonType;
+            if (!controller.hasOutgoingRequest() && row.canConnectToNode) {
+              buttonType = 'accept-button';
+            }
+            var incomingStatus = i18n.lobbyStatusWaitingForYou();
+            writeNodeRow(row.nodeID, row.displayName, incomingStatus, buttonType, row.classAttr);
+          });
+        }
+
+        // Outgoing request table
+        if (outgoingRequestRows.length > 0) {
+          writeEmptyRow();
+          writeHeader(i18n.outgoingConnectionRequests());
+          outgoingRequestRows.forEach(function (row) {
+            var outgoingStatus = i18n.lobbyStatusWaitingForOther({
+              spinner: '<img src="/assets/spinner-big.gif" />',
```

Comment:
grunt tests OK with this? I think we only use netsim from within the dashboard context so this should prob. always be available, just worth keeping an eye out